tests/test_packages/test_connections/test_p2p_libp2p_client/test_errors.py

```diff
@@ -28,104 +28,160 @@
 import pytest
 
 from aea.configurations.base import ConnectionConfig
-from aea.configurations.constants import DEFAULT_LEDGER
-from aea.crypto.registries import make_crypto
-from aea.helpers.base import CertRequest
-from aea.identity.base import Identity
-from aea.multiplexer import Multiplexer
-
-from packages.valory.connections.p2p_libp2p.consts import (
-    LIBP2P_CERT_NOT_AFTER,
-    LIBP2P_CERT_NOT_BEFORE,
-)
+
 from packages.valory.connections.p2p_libp2p_client.connection import (
     NodeClient,
     P2PLibp2pClientConnection,
-    POR_DEFAULT_SERVICE_ID,
 )
 
 from tests.test_packages.test_connections.test_p2p_libp2p.base import (
     BaseP2PLibp2pTest,
-    LIBP2P_LEDGER,
     _make_libp2p_client_connection,
     _make_libp2p_connection,
     _process_cert,
+    create_identity,
     libp2p_log_on_failure_all,
+    make_cert_request,
     ports,
 )
 
 
+DONE_FUTURE: asyncio.Future = asyncio.Future()
+DONE_FUTURE.set_result(None)
+
+
 @pytest.mark.asyncio
-class TestLibp2pClientConnectionFailureNodeNotConnected:
+class TestLibp2pClientConnectionFailureNodeNotConnected(BaseP2PLibp2pTest):
     """Test that connection fails when node not running"""
 
+    public_key = BaseP2PLibp2pTest.default_crypto.public_key
+    connection = _make_libp2p_client_connection(peer_public_key=public_key)
+
```

Comment:
I grouped async tests from this module here (were single functions). Reuse the class to inherit at mailbox module to reuse these tests there